### PR TITLE
Load balance Admin API

### PR DESCRIPTION
**Pull request by schananas**
Exposes load balance API

### CODE REVIEW

File: src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java [hunk 1]
```diff
@@ -107,6 +107,20 @@
...
+    /**
+     * Request to balance the load for the specified event processor among the connected client.
+     * Returns a {@link CompletableFuture} that completes when the request has been received by AxonServer.
+     * This doesn't imply that the processor have been balanced, but only that the request has been properly
+     * delivered.
+     *
+     * @param eventProcessorName   the name of the event processor to merge
+     * @param tokenStoreIdentifier the token store identifier of the processor to merge
+     * @param strategy             the balancing strategy to use
+     * @return a {@link CompletableFuture} that completes when the request has been delivered to AxonServer
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java:119]: Got some rewording suggestions for the Javadoc:

```suggestion
     * Request to balance the load for the given {@code eventProcessorName} within the connected client.
     * Returns a {@link CompletableFuture} that completes when the request has been received by Axon Server.
     * Note that this doesn't imply that the processor is balanced, but only that the request has been properly
     * delivered.
     *
     * @param eventProcessorName   the name of the event processor to balance the load for
     * @param tokenStoreIdentifier the token store identifier of the processor to balance the load for
     * @param strategy             the balancing strategy to use
     * @return a {@link CompletableFuture} that completes when the request is delivered to Axon Server
```

File: src/main/java/io/axoniq/axonserver/connector/admin/LoadBalanceStrategy.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package io.axoniq.axonserver.connector.admin;
+
+/**
+ * @author Stefan Dragisic
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/LoadBalanceStrategy.java:4]: The JavaDoc of this class is incomplete.
I expect the following still:

* A class level description
* A description per enumeration option
* A description for *any* public method

File: src/main/java/io/axoniq/axonserver/connector/admin/LoadBalanceStrategy.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public io.axoniq.axonserver.grpc.admin.LoadBalanceStrategy toGrpc() {
+        switch (this) {
+            case DEFAULT : return io.axoniq.axonserver.grpc.admin.LoadBalanceStrategy.DEFAULT;
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/LoadBalanceStrategy.java:23]: I'd wager @MGathier his point on not using an enumeration in the axonserver-api will have some impact on this method.

File: src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java [hunk 4]
```diff
@@ -168,6 +170,18 @@ protected EventProcessor terminalMessage() {
         });
     }
 
+    @Override
+    public CompletableFuture<Void> loadBalanceEventProcessor(String eventProcessorName, String tokenStoreIdentifier,
+                                                             LoadBalanceStrategy strategy) {
+        EventProcessorIdentifier eventProcessorIdentifier = eventProcessorId(eventProcessorName, tokenStoreIdentifier);
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
+        eventProcessorServiceStub.loadBalanceProcessor(LoadBalanceRequest.newBuilder().setProcessor(eventProcessorIdentifier)
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java:178]: Nit: the indentation seems to be out of sync with the style guide.

File: src/main/java/io/axoniq/axonserver/connector/admin/LoadBalanceStrategy.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package io.axoniq.axonserver.connector.admin;
+
+/**
+ * @author Stefan Dragisic
+ */
+public enum LoadBalanceStrategy {
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/LoadBalanceStrategy.java:6]: I am missing test cases for the public static methods of this enumeration.

**Review by smcvb — CHANGES_REQUESTED** [review]
Couple of pointers here, like JavaDoc, tests, and the use of the enumeration in general.

File: src/test/java/io/axoniq/axonserver/connector/admin/AdminChannelIntegrationTest.java [hunk 6]
```diff
@@ -45,7 +45,7 @@
  * @author Sara Pellegrini
  * @since 4.6.0
  */
-class AdminChannelIntegrationTest extends AbstractAxonServerIntegrationTest {
+class AdminChannelIntegrationTest  extends AbstractAxonServerIntegrationTest {
```
Comment (smcvb) [inline src/test/java/io/axoniq/axonserver/connector/admin/AdminChannelIntegrationTest.java:48]: I am missing a test for the `getBalancingStrategies` method you introduced.

File: src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java [hunk 7]
```diff
@@ -168,6 +170,39 @@ protected EventProcessor terminalMessage() {
         });
     }
 
+    @Override
+    public CompletableFuture<Void> loadBalanceEventProcessor(String eventProcessorName, String tokenStoreIdentifier,
+                                                             String strategy) {
+        EventProcessorIdentifier eventProcessorIdentifier = eventProcessorId(eventProcessorName, tokenStoreIdentifier);
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
+        eventProcessorServiceStub.loadBalanceProcessor(LoadBalanceRequest.newBuilder().setProcessor(
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java:178]: Nit: indentation seems off to me here.

File: src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java [hunk 8]
```diff
@@ -168,6 +170,39 @@ protected EventProcessor terminalMessage() {
...
+        return responseObserver.thenRun(() -> {
+        });
+    }
+
+    @Override
+    public CompletableFuture<Void> autoLoadBalanceEventProcessor(String eventProcessorName, String tokenStoreIdentifier,
+                                                             String strategy) {
+        EventProcessorIdentifier eventProcessorIdentifier = eventProcessorId(eventProcessorName, tokenStoreIdentifier);
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
+        eventProcessorServiceStub.autoLoadBalanceProcessor(LoadBalanceRequest.newBuilder().setProcessor(
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java:191]: Nit: indentation seems off to me here.

File: src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java [hunk 9]
```diff
@@ -168,6 +170,39 @@ protected EventProcessor terminalMessage() {
...
+                                                                                 eventProcessorIdentifier)
+                                                                         .setStrategy(strategy)
+                                                                         .build(), responseObserver);
+        return responseObserver.thenRun(() -> {
+        });
+    }
+
+    @Override
+    public CompletableFuture<Void> autoLoadBalanceEventProcessor(String eventProcessorName, String tokenStoreIdentifier,
+                                                             String strategy) {
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java:188]: Nit: indentation seems off to me here.

File: src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java [hunk 10]
```diff
@@ -107,6 +108,40 @@
...
+     */
+    CompletableFuture<Void> autoLoadBalanceEventProcessor(String eventProcessorName, String tokenStoreIdentifier, String strategy);
+
+
+    /**
+     * Returns all available load balance strategies registered to AxonServer.
+     *
+     * @return the list of all available load balance strategies registered to AxonServer.
+     */
+    CompletableFuture<List<LoadBalancingStrategy>> getBalancingStrategies();
```
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java:143]: I can make assumption here, but I think it's better to ask: how are you going to use this method exactly?

File: src/test/java/io/axoniq/axonserver/connector/admin/AdminChannelIntegrationTest.java (hunk 6 above)
Comment (schananas) [reply src/test/java/io/axoniq/axonserver/connector/admin/AdminChannelIntegrationTest.java:48]: It is there already:

```
     CompletableFuture<Void> accepted = adminChannel.getBalancingStrategies()
                .thenCompose(balancingStrategies -> adminChannel.loadBalanceEventProcessor("processor", "tokenStore", balancingStrategies.get(0).getStrategy()));
```

File: src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java (hunk 10 above)
Comment (schananas) [reply src/main/java/io/axoniq/axonserver/connector/admin/AdminChannel.java:143]: Its in the test:

```
     CompletableFuture<Void> accepted = adminChannel.getBalancingStrategies()
                .thenCompose(balancingStrategies -> adminChannel.loadBalanceEventProcessor("processor", "tokenStore", balancingStrategies.get(0).getStrategy()));
```

**Review by smcvb — APPROVED** [review]
My concerns have been addressed, hence I'm approving this pull request.

**Pull request merged**